doc/source/changelog.md

```diff
@@ -2,34 +2,64 @@
 
 ## Unreleased
 
+## v1.4.0 (2023-03-31)
+
+### Thanks to our contributors
+
+We would like to give our special thanks to all the contributors who made the new version of Flower possible (in `git shortlog` order):
+
+`Adam Narozniak`, `Charles Beauville`, `Chenyang Ma (Danny)`, `Daniel J. Beutel`, `Gautam Jajoo`, `Iacob-Alexandru-Andrei`, `Kunal Sarkhel`, `Max Kapsecker`, `Michał`, `Nic Lane`, `Nikolaos Episkopos`, `Semo Yang`, `Steve Laskaridis`, `Taner Topal`
+
```

Comment:
Let's remove this part for now and only add it just before the release to make sure we're not missing any contributors we might add in between today and the actual release date.